### PR TITLE
Capture console output from tasks

### PR DESCRIPTION
**Pull request by davisagli**
Note: for now, this is only enabled when the CAPTURE_TASK_OUTPUT env var is set.

# Critical Changes

# Changes
- Output from tasks to stdout or stderr is now captured and included in CumulusCI logs. [Wait to include this in the release notes until we turn it on by default.]

# Issues Closed

### CODE REVIEW

**Review by jstvz — APPROVED** [review]

File: .github/workflows/feature_test.yml [hunk 1]
```diff
@@ -56,6 +56,8 @@ jobs:
                   git fetch --depth=1 --no-tags origin $GITHUB_BASE_REF:$GITHUB_BASE_REF
                   git fetch --depth=1 --no-tags origin $GITHUB_HEAD_REF:$GITHUB_HEAD_REF
                   pre-commit run --files $(git diff --diff-filter=d --name-only $GITHUB_BASE_REF..$GITHUB_HEAD_REF)
+              env:
+                  SKIP: debug-statements
```
[the remark this answers is not in the document]
Comment (davisagli) [reply .github/workflows/feature_test.yml:60]: @jstvz yes, that sounds better

File: cumulusci/core/tasks.py [hunk 2]
```diff
@@ -18,10 +18,19 @@
...
+from cumulusci.utils.logging import redirect_output_to_logger
 
 CURRENT_TASK = threading.local()
 
 PROJECT_CONFIG_RE = re.compile(r"\$project_config.(\w+)")
+CAPTURE_TASK_OUTPUT = os.environ.get("CAPTURE_TASK_OUTPUT")
+
+
+# We can't use contextlib.nullcontext yet
+# because it isn't present in Python 3.6
```
[the remark this answers is not in the document]
Comment (davisagli) [reply cumulusci/core/tasks.py:30]: Respectfully disagree; I think it's just as likely we'll search for "Python 3.6" when we remove 3.6 support. FIXME implies it's broken now, which it's not.

File: cumulusci/core/tests/test_tasks.py [hunk 3]
```diff
@@ -30,23 +31,27 @@ def _run_task(self):
         return -1
 
 
-class TestBaseTaskCallable(MockLoggerMixin, unittest.TestCase):
```
[the remark this answers is not in the document]
Comment (davisagli) [reply cumulusci/core/tests/test_tasks.py:33]: The dumb way, and then I started thinking that such a tool should exist but didn't actually look for it. Thanks for the pointer for next time.

File: .github/workflows/feature_test.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (davisagli) [reply .github/workflows/feature_test.yml]: https://github.com/SFDO-Tooling/CumulusCI/pull/2827/commits/1e0cce5f0c9e3b62f43fae0380267ff704304874

File: cumulusci/core/tasks.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jstvz) [reply cumulusci/core/tasks.py:30]: >> **Add** a `FIXME` for later?
> Respectfully disagree;  I think it's just as likely we'll search for "Python 3.6" when we remove 3.6 support. 

😏 
> FIXME implies it's broken now, which it's not.

Fair enough.

**Review by jstvz — APPROVED** [review]
Integration test failure doesn't _look_  meaningful (the output is hard to parse), approving.
